### ISSUE TITLE
PermissionError when running `clear`

### ISSUE THREAD

**Issue by fxfitz** [xonsh/xonsh#1051]
```
$ clear





















Traceback (most recent call last):
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/base_shell.py", line 154, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/codecache.py", line 57, in run_compiled_code
    func(code, glb, loc)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xontrib/prompt_ret_code.xsh", line 1, in <module>
    from xonsh.tools import ON_WINDOWS
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/built_ins.py", line 671, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/built_ins.py", line 573, in run_subproc
    wait_for_active_job()
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 117, in wait_for_active_job
    _continue(active_task)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 58, in _continue
    _send_signal(job, signal.SIGCONT)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 64, in _send_signal
    os.killpg(job['pgrp'], signal)
PermissionError: [Errno 1] Operation not permitted
```

**Comment by smogg** [comment]
+1, second `clear` also kills my terminal

**Comment by fxfitz** [comment]
Actually, this appears to be sporadic. :'(

**Comment by adqm** [comment]
Weird!  Thanks for reporting, and thanks for including the traceback.  I haven't been able to reproduce this, but maybe one of the developers running Mac can.

Is this happening for all commands, or just `clear`?

**Comment by smogg** [comment]
it just killed my terminal while using `echo` so it appears it's for all commands. I was not fast enough to read the traceback

**Comment by fxfitz** [comment]
Here's my echo:
```
$ echo hi
hi
Traceback (most recent call last):
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/base_shell.py", line 154, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/codecache.py", line 57, in run_compiled_code
    func(code, glb, loc)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xontrib/prompt_ret_code.xsh", line 1, in <module>
    from xonsh.tools import ON_WINDOWS
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/built_ins.py", line 671, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/built_ins.py", line 573, in run_subproc
    wait_for_active_job()
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 117, in wait_for_active_job
    _continue(active_task)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 58, in _continue
    _send_signal(job, signal.SIGCONT)
  File "/Users/fxfitz/.pyenv/versions/3.5.0/lib/python3.5/site-packages/xonsh/jobs.py", line 64, in _send_signal
    os.killpg(job['pgrp'], signal)
PermissionError: [Errno 1] Operation not permitted
```

**Comment by sehqlr** [comment]
I just installed xonsh after watching the PyCon 2016 talk (today), and this is happening to me as well. I don't know if this extra info helps, but for me, after the PermissionError, the prompt completely hangs, and the `clear` process resists `SIGKILL`.

**Comment by adqm** [comment]
@sehqir, are you on Mac El Capitan as well?

**Comment by sehqlr** [comment]
```
+------------------+-------+
| xonsh            | 0.3.2 |
| Python           | 3.5.1 |
| PLY              | 3.7   |
| have readline    | True  |
| prompt toolkit   | None  |
| pygments         | 2.1.3 |
| on posix         | True  |
| on linux         | False |
| on darwin        | True  |
| on windows       | False |
| is superuser     | False |
| default encoding | utf-8 |
+------------------+-------+
```

**Comment by adqm** [comment]
Alright; we've not forgotten about you @sehqlr, @fxfitz, @smogg!  It looks like the way OSX is handling process groups is different from what we expected.

@Carreau described a partial workaround in #1012, so we'll keep exploring that.

**Comment by moigagoo** [comment]
Other commands crash as well. In particular, rm, clear, ping without arguments. As mentioned above, the crashes are random.

**Comment by moigagoo** [comment]
I can't reproduce the issue. Looks like it's fixed! Thanks a lot!

**Comment by adqm** [comment]
Great!

**Comment by adqm** [comment]
I believe this was fixed with #1106, so I'm going to close this issue for now.  If it persists, please feel free to reopen!
